tools/system_libs.py

```diff
@@ -1171,6 +1200,11 @@ def add_library(lib):
     force_include.add('liblsan_rt_wasm')
     add_library(system_libs_map['liblsan_rt_wasm'])
 
+  if shared.Settings.PROXY_POSIX_SOCKETS:
+    add_library(system_libs_map['libsockets_proxy'])
+  else:
+    add_library(system_libs_map['libsockets'])
```

Comment:
Is it a good idea to always link sockets? This will include the libraries unconditionally. If you want the library to be used when needed, use a symbol file.

Reply:
libc with sockets has been linked in unconditionally from a long time ago (musl has sockets code built-in), this PR does not seek to change that model. Including the libraries unconditionally does not add to code size if the target project does not use any of the sockets symbols.

(I do not oppose moving to requiring e.g. `-lsocket` to get sockets code, but don't want to work on changing that model here)